controller/generic/metacontroller.go

```diff
@@ -290,38 +322,56 @@ func (mc *ConfigMetaController) wait(condition func() (bool, error)) error {
 // binary
 //
 // NOTE:
-//	This method is used as a condition and hence can be invoked
-// more than once.
+//	This method is used as a condition and is repeatedly executed
+// under a loop till this condition is not met.
 func (mc *ConfigMetaController) startAllWatchControllers() (bool, error) {
-	// In this metacontroller, we are only responsible for
-	// starting/stopping the relevant watch based controllers
+	var errs []string
+	// This logic is responsible to start all the generic controllers
+	// configured in the config file
 	for _, conf := range mc.Configs {
```

Comment:
I do not know threading in go, so maybe couple questions here :
* if we have two controllers, one run fine, and second have problems, what happens on second loop ? Does the first one is killed and started again ? Could it happen that controller with erros started new threads and they should be cleaned before starting again ?

Reply:
- The first one will not be retried in second loop
- First one will never be killed in second loop
- Controllers that error out are cleaned up before as soon as they hit errors